src/components/buttons/z-toggle-switch/index.tsx

```diff
@@ -0,0 +1,86 @@
+import { Component, Prop, h, Event, EventEmitter } from "@stencil/core";
+import { randomId } from "../../../utils/utils";
+import { keybordKeyCodeEnum } from "../../../beans";
+
+@Component({
+  tag: "z-toggle-switch",
+  styleUrl: "styles.css",
+  shadow: false,
+  scoped: true,
+})
+export class ZToggleSwitch {
+  @Prop({ reflect: true }) disabled?: boolean = false;
+  @Prop({ reflect: true }) reverselabel?: boolean = false;
+  @Prop({ mutable: true }) checked?: boolean = false;
+  @Prop() htmlid: string = `id-${randomId()}`;
+
+  @Event() toggleClick: EventEmitter;
+  emitToggleClick(checked: boolean) {
+    this.toggleClick.emit({
+      id: this.htmlid,
+      checked: checked,
+    });
+  }
+
+  handleClick(ev) {
+    this.checked = ev.target.checked;
+    this.emitToggleClick(this.checked);
+  }
+
+  handleKeyboardEvent(ev) {
+    if (
+      ev.keyCode === keybordKeyCodeEnum.ENTER ||
+      ev.keyCode === keybordKeyCodeEnum.SPACE
+    ) {
+      this.checked = !this.checked;
+      this.emitToggleClick(this.checked);
+    }
+  }
+
+  render() {
+    return [
+      <label
+        htmlFor={this.htmlid}
+        class={{
+          left: !this.reverselabel,
+          right: this.reverselabel,
+        }}
+      >
+        <span
+          class={{
+            left: !this.reverselabel,
+            right: this.reverselabel,
+          }}
+        >
+          <slot />
+        </span>
+        <div
+          tabIndex={0}
+          class={`container ${this.disabled && "disabled"} ${
+            this.checked && "active"
+          }`}
+          onKeyDown={(e: KeyboardEvent) => this.handleKeyboardEvent(e)}
```

Comment:
Utilizzerei `onKeyDown` invece di `onKeyUp`, e sfrutterei il metodo `handleKeyboardSubmit` presente in utils, passandogli come callback la `this.handleClick`.
Aggiungerei il controllo sul disabled direttamente dentro la handleClick, così è valido per tutti e non dovrebbe creare problemi.